src/sage/combinat/k_regular_sequence.py

```diff
@@ -918,6 +979,367 @@ def _n_to_index_(self, n):
 
+    def guess(self, f, n_max=100, max_exponent=10, sequence=None):
+        r"""
+        Guess a `k`-regular sequence whose first terms coincide with `(f(n))_{n\geq0}`.
+
+        INPUT:
+
+        - ``f`` -- a function (callable) which determines the sequence.
+          It takes nonnegative integers as an input
+
+        - ``n_max`` -- (default: ``100``) a positive integer. The resulting
+          `k`-regular sequence coincides with `f` on the first ``n_max``
+          terms
+
+        - ``max_exponent`` -- (default: ``10``) a positive integer specifying
+          the maximum exponent of `k` which is tried when guessing the sequence
+
+        - ``sequence`` -- (default: ``None``) a `k`-regular sequence used
+          for bootstrapping the guessing by adding information of the
+          linear representation of ``sequence`` to the guessed representation
+
+        OUTPUT:
+
+        A :class:`kRegularSequence`
+
+        ALGORITHM:
+
+        For the purposes of this description, the left vector valued sequence
+        associated with a regular sequence is consists of the left vector
+        multiplied by the corresponding matrix product, but without the right
+        vector of the regular sequence.
+
+        The algorithm maintains a left vector valued sequence consisting
+        of the left vector valued sequence of the argument ``sequence``
+        (replaced by an empty tuple if ``sequence`` is `None`) plus several
+        components of the shape `m \mapsto f(k^t\cdot m +r)` for suitable
+        ``t`` and ``r``.
+
+        Implicitly, the algorithm also maintains a `d \times n_max` matrix ``A``
+        (where ``d`` is the dimension of the left vector valued sequence)
+        whose columns are the current left vector valued sequence evaluated at
+        the non-negative integers less than `n_max` and ensures that this
+        matrix has full row rank.
+
+        EXAMPLES:
+
+        Binary sum of digits::
+
+            sage: @cached_function
+            ....: def s(n):
+            ....:     if n == 0:
+            ....:         return 0
+            ....:     return s(n//2) + ZZ(is_odd(n))
+            sage: all(s(n) == sum(n.digits(2)) for n in srange(10))
+            True
+            sage: [s(n) for n in srange(10)]
+            [0, 1, 1, 2, 1, 2, 2, 3, 1, 2]
+
+        Let us guess a `2`-linear representation for `s(n)`::
+
+            sage: Seq2 = kRegularSequenceSpace(2, ZZ)
+            sage: import logging
+            sage: logging.basicConfig(level=logging.INFO)
+            sage: S1 = Seq2.guess(s); S1
+            INFO:...:including f_{1*m+0}
+            INFO:...:including f_{2*m+1}
+            2-regular sequence 0, 1, 1, 2, 1, 2, 2, 3, 1, 2, ...
+            sage: S1.linear_representation()
+            ((1, 0),
+             Finite family {0: [1 0]
+                               [0 1],
+                            1: [ 0  1]
+                               [-1  2]},
+             (0, 1))
```

Comment:
Following https://github.com/sagemath/sage/pull/35682/files#r1205543046, I suggest the following.
```suggestion
             (0, 1))
             
         The ``INFO`` messages mean that the right vector valued sequence is the sequence `(s(n), s(2n+1))^\top`.
```